exporter/cortexexporter/cortex.go

```diff
@@ -0,0 +1,354 @@
+	default:
+		// stores TimeSeries
+		tsMap := map[string]*prompb.TimeSeries{}
+		dropped := 0
+		errs := []string{}
+
+		rms := data.MetricDataToOtlp(pdatautil.MetricsToInternalMetrics(md))
+
+		for _, r := range rms {
+			// TODO: add resource attributes as labels
+			for _, inst := range r.InstrumentationLibraryMetrics {
+				//TODO: add instrumentation library information as labels
+				for _, m := range inst.Metrics {
+					// check for valid type and temporality combination
+					ok := validateMetrics(m.MetricDescriptor)
+					if !ok {
+						dropped++
+						errs = append(errs, "invalid temporality and type combination")
+						continue
+					}
+					// handle individual metric based on type
+					switch m.GetMetricDescriptor().GetType() {
+					case otlp.MetricDescriptor_MONOTONIC_INT64, otlp.MetricDescriptor_INT64,
+						otlp.MetricDescriptor_MONOTONIC_DOUBLE, otlp.MetricDescriptor_DOUBLE:
+						if err := ce.handleScalarMetric(tsMap, m); err != nil {
+							errs = append(errs, err.Error())
+						}
+					case otlp.MetricDescriptor_HISTOGRAM:
+						if err := ce.handleHistogramMetric(tsMap, m); err != nil {
+							errs = append(errs, err.Error())
+						}
+					case otlp.MetricDescriptor_SUMMARY:
+						if err := ce.handleSummaryMetric(tsMap, m); err != nil {
+							errs = append(errs, err.Error())
+						}
+					default:
+						dropped++
+						errs = append(errs, "invalid type")
+						continue
+					}
+				}
+			}
+		}
+
+		if err := ce.export(ctx, tsMap); err != nil {
+			return pdatautil.MetricCount(md), err
+		}
+
+		if dropped != 0 {
+			return dropped, fmt.Errorf(strings.Join(errs, "\n"))
+		}
+
+		return 0, nil
+	}
+}
+
+// handleScalarMetric processes data points in a single OTLP scalar metric by adding the each point as a Sample into
+// its corresponding TimeSeries in tsMap.
+// tsMap and metric cannot be nil, and metric must have a non-nil descriptor
+func (ce *cortexExporter) handleScalarMetric(tsMap map[string]*prompb.TimeSeries, metric *otlp.Metric) error {
+
+	mType := metric.MetricDescriptor.Type
+
+	switch mType {
+	// int points
+	case otlp.MetricDescriptor_MONOTONIC_INT64, otlp.MetricDescriptor_INT64:
+		if metric.Int64DataPoints == nil {
+			return fmt.Errorf("nil data point field in metric" + metric.GetMetricDescriptor().Name)
+		}
+
+		for _, pt := range metric.Int64DataPoints {
+
+			// create parameters for addSample
+			name := getPromMetricName(metric.GetMetricDescriptor(), ce.namespace)
+			lbs := createLabelSet(pt.GetLabels(), nameStr, name)
+			sample := &prompb.Sample{
+				Value:     float64(pt.Value),
+				Timestamp: int64(pt.TimeUnixNano),
+			}
+
+			addSample(tsMap, sample, lbs, metric.GetMetricDescriptor().GetType())
+		}
+		return nil
+
+	// double points
+	case otlp.MetricDescriptor_MONOTONIC_DOUBLE, otlp.MetricDescriptor_DOUBLE:
+		if metric.DoubleDataPoints == nil {
+			return fmt.Errorf("nil data point field in metric" + metric.GetMetricDescriptor().Name)
```

Comment:
`fmt.Errorf` and `errors.Errorf` are both being used. I think it would be best to stick with one for everything.